src/test/analyzer_test.ts

```diff
@@ -523,4 +524,63 @@ suite('Analyzer', () => {
           testSeed.events.map(e => e.name), ['fired-event', 'data-changed']);
     });
   });
+
+  suite.skip('race conditions and caching', () => {
+    const wait = () =>
+        new Promise((resolve) => setTimeout(resolve, Math.random() * 30));
```

Comment:
Yeah, I hear you. I think this is still worthwhile to have around though, as a stress test of unexpected routes through our analysis and caching pipeline.

How about I add deterministic tests of all the three failures states that I noticed this test encountering?